### PR TITLE
Initial working version

### PR DESCRIPTION
**Pull request by ritch**
This is an extraction of the loopback filter mechanism from the juggler's memory connector. It supports the suite of tests from the memory connector. Specifically:
- `where`
- `limit`
- `skip` (offset)
- `fields`

**It does not support `include` or input validation.**

/to @raymondfeng 
/cc @bajtos @fabien

**Basic Example** (could use some ideas for better apis)

``` js
var data = [{foo: 'bar'}, {bat: 'baz'}, {foo: 'bar'}];
var filter = {where: {foo: 'bar'}};
var filtered = require('loopback-filter')(data, filter);
```

### CODE REVIEW

**Comment by bajtos** [comment]
awesome :+1:

**Comment by rmg** [comment]
@slnode test please

**Comment by ritch** [comment]
test please

File: index.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
+var debug = require('debug')('loopback:filter');
+var geo = require('./lib/geo');
+
+module.exports = function filterNodes(nodes, filter) {
+  if (filter) {
```
Comment (raymondfeng) [inline index.js:5]: Should we assert/normalize the args as follows?

nodes --> []
filter --> {}

File: index.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  var where = filter.where;
+  if (typeof where === 'function') {
+    return where;
+  }
+  return function(obj) {
+    return matchesFilter(obj, filter);
+  }
+}
+
+function matchesFilter(obj, filter) {
```
Comment (raymondfeng) [inline index.js]: It would be nice to export a `matches(obj, where)` function.

Comment (ritch) [inline index.js]: good idea

File: index.js (hunk 1 above)
Comment (ritch) [inline index.js:5]: Ya I'll create an issue to talk about sanitization. I don't think we need it for `v0.1.0`.

**Comment by ritch** [comment]
test please

**Comment by raymondfeng** [comment]
LGTM

**Pull request merged**